modules/au.com.trgtd.tr.task.messages/test/unit/src/au/com/trgtd/tr/task/messages/MessageParserTest.java

```diff
@@ -0,0 +1,126 @@
+package au.com.trgtd.tr.task.messages;
+
+import au.com.trgtd.tr.task.messages.MessageParser.Message;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Date;
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.List;
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+public class MessageParserTest {
+
+    private MessageParser parser = new MessageParser();
+    private LocalDate localDate = LocalDate.of(2014, Month.MARCH, 26);
+
+    final String xml1 = "<msgs>"
+            + "<msg date='2014/03/27' type='A'>message1</msg>"
+            + "<msg date='2014/03/30' type='B'>message2</msg>"
+            + "</msgs>";
+
+    final String xml2 = "<msgs>"
+            + "<msg date='2014/03/27' type='N'>nonMemberMessage</msg>"
+            + "<msg date='2014/03/30' type='M'>memberMessage</msg>"
+            + "</msgs>";
+
+    @Test
+    public void parsingValidButEmptyXml_returnsNoMessages() throws Exception {
+        assertTrue(parse("<msgs/>", localDate).isEmpty());
+    }
+
+    @Test
+    public void parsingXml1_withDateBeforeBoth_returnsBothMessages() throws Exception {
```

Comment:
In dependency of provious change this should be
parsingSampleMessage1_witchDateBeforeBoth_returnsBoothMessages 
etc.

Reply:
Had to get off the train, before i could reread all. Will follow up, thanks